docs/rules/no-keyword-prefix.md

```diff
@@ -0,0 +1,20 @@
+# Disallow identifiers starting with `new` or `class`
+
+`new Foo` and `newFoo` look very similar. Use alternatives that do not look like keyword usage.
+
+## Fail
+
+```js
+const newFoo = 'foo';
+const classFoo = 'foo';
+```
+
+
+## Pass
+
+```js
+const foo = 'foo';
+const _newFoo = 'foo';
+const new_foo = 'foo';
+const fooNew = 'foo';
+```
```

Comment:
Would it make sense to let users configure prefixes to disallow? To make the rule more generally useful.

Reply:
It wouldn't be hard to add. I liked having "keyword" in the rule name because that was my reasoning for what should be disallowed so then I limited its usage to the keywords used in a camel case context. But having it be customizable is fine.

(On the other hand, I doubt many other people would are going to use the rule...)

Reply:
Yeah, let's keep it simple for now. I can't think of any use-cases for making it configurable.